p2pp/mcf.py

```diff
@@ -677,14 +680,13 @@ def gcode_parseline(index):
     if v.side_wipe and block_class == CLS_NORMAL and classupdate:
         create_side_wipe()
 
-    # g.Comment = " ; - {}".format(v.total_material_extruded)
-    g.issue_command()
-
-    if (g.X or g.Y) and v.retracted:
+    if (g.X or g.Y) and g.E > 0 and v.retracted:
```

Comment:
There are cases where multiple non-extrude movements were seen in the gcode, placing this before the issue_command and checking E makes this a bit more flexible.